dnf/base.py

```diff
@@ -2153,11 +2153,18 @@ def _prov_key_data(msg):
                     continue
 
                 # Import the key
+                # If rpm.RPMTRANS_FLAG_TEST in self._ts, gpg keys cannot be imported successfully
+                test_flag = self._ts.isTsFlagSet(rpm.RPMTRANS_FLAG_TEST)
+                if test_flag:
+                    orig_flags = self._ts.getTsFlags()
+                    self._ts.setFlags(orig_flags - rpm.RPMTRANS_FLAG_TEST)
```

Comment:
I'd prefer a more thread-safe approach: don't modify self._ts, but create a new ts with modified flags to cover this scenario. Is it all about the flags or are there any other ts attributes that have to be set?